### DISCUSSION TITLE
v9.1.0 My structure does not implement interface with []*redis.Client (missing method BFAdd)

### DISCUSSION THREAD

**Discussion by suuvor** [redis/go-redis#2691] [General]
Hello,

I see the next error after go-redis upgrade from v9.0.5 to v9.1.0
```
cannot use &myStruct{…} (value of type *myStruct) as DB value in return statement: *myStruct does not implement DB (missing method BFAdd)
```

In go.mod I have the right version of redis:
```
require (
    ...
    github.com/redis/go-redis/v9 v9.1.0
)
```

How client is called:
```
myStruct struct {
    urls         []string
    instances    []*redis.Client
}

DB interface {
    redis.Cmdable
}

func MustConnect(...) DB {
    for _, url := range URLs {
        opts, err := redis.ParseURL(url)
        client := redis.NewClient(opts)
        clients = append(clients, client)
    }

    return &myStruct{instances: clients, urls: urls}
}
```
What is wrong now with v9.1.0 version?

**Comment by monkey92t** [comment]
From your code, it appears that &myStruct doesn't implement the DB interface. This is unrelated to go-redis and has nothing to do with either v9.1.0 or v9.0.5.

**Reply by suuvor** [reply]
It's related to redis upgrade from v9.0.5 to v9.1.0 and redis-go interface change. It was added
```
type probabilisticCmdable interface {
```
And this interface is private.
Do you have suggestions how to overcome this now?
